Assets/Scripts/Source/Audio/AudioManager.cs

```diff
@@ -32,10 +32,61 @@ public void Play(string sourceName, AudioClip clip, AudioOptions options=null)
         {
             options = AudioOptions.GetDefault();
         }
+        if (clip == null)
+        {
+            clip = source.clip;
+        }
+        if (clip == null)
+        {
+            Debug.LogErrorFormat("Clip {0} not found.", sourceName);
+            return;
+        }
         clipOptions[getClipIdentifier(source, clip)] = options;
         handleClipTransition(source, clip, options);
     }
 
+    public void Stop(string sourceName, AudioClip clip, float fadeOutTime = 0f)
+    {
+        AudioSource source;
+        if (!sources.TryGetValue(sourceName, out source))
+        {
+            Debug.LogErrorFormat("Source {0} not found. Unable to stop clip.", sourceName);
+            return;
+        }
+        if (clip == null)
+        {
+            clip = source.clip;
+        }
+        if (clip == null)
+        {
+            Debug.LogErrorFormat("Clip {0} not found.", sourceName);
+            return;
+        }
+
+        if (!clipOptions.ContainsKey(getClipIdentifier(source, clip)))
+        {
+            Debug.LogErrorFormat("Source {0} options not found. Unable to stop clip.", sourceName);
+            return;
+        }
+
+        AudioOptions options = clipOptions[getClipIdentifier(source, clip)];
+        Action endHandlerAction = delegate { options.OnEnd(); };
+
+        options.FadeOutTime = fadeOutTime;
```

Comment:
I'm not sure if it makes sense to permanently overwrite the `AudioOptions` fade out time for this clip. Seems better to keep the local fade out time scoped to the function